Infrastructure/GhostNetwork.Profiles.MongoDb/SecuritySettingsStorage.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Threading.Tasks;
+using GhostNetwork.Profiles.SecuritySettings;
+using MongoDB.Driver;
+
+namespace GhostNetwork.Profiles.MongoDb
+{
+    public class SecuritySettingsStorage : ISecuritySettingStorage
+    {
+        private readonly MongoDbContext context;
+
+        public SecuritySettingsStorage(MongoDbContext context)
+        {
+            this.context = context;
+        }
+
+        public async Task UpdateAsync(Guid userId, SecuritySetting updatedSettings)
+        {
+            var filter = Builders<SecuritySettingsEntity>.Filter.Eq(x => x.UserId, userId);
+            var settings = await context.SecuritySettings.Find(filter).FirstOrDefaultAsync();
```

Comment:
You can use UpdateOneAsync to avoid extra code for inserting
`await collection.UpdateOneAsync(filter, update, new UpdateOptions() { IsUpsert = true })`